presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -283,6 +284,11 @@ public SystemSessionProperties(
                         featuresConfig.isLegacyOrderBy(),
                         false),
                 booleanSessionProperty(
+                        SUPPORT_NULL_ON_LHS_OF_SEMI_JOIN,
```

Comment:
@sopel39 in your example, can you somehow push the predicate down _correctly_? Maybe if you knew `x` cannot be null, then `x IN (select y from very_large_table where y=3 or y is null)`, but currently we don't have nullability  in our type system.